classes/fsl-eula.bbclass

```diff
@@ -0,0 +1,29 @@
+# fsl-eula.bbclass defines a common naming scheme used by the
```

Comment:
I really dislike this rework of the class. It puts a lot of things hidden and it is hard for me to buy the idea of this extra complexity.

Couldn't we use the class we have now?

Reply:
This is not a rework, this is an extension. The existing class is inherited, and the existing recipes will continue to use the existing class as-is.

The existing class does not handle the need to produce multiple binaries along multiple axes like architecture and graphics backend. Rather than duplicate the code for that in multiple recipes, I chose to capture it in these classes for maximum reuse and actually for maximum hiding, as that is the point of abstraction. All the recipes using these classes are much simpler.

You're not seeing the full usage of the extension, as that is from a demo layer we created:
https://source.codeaurora.org/external/imx/meta-imx-gpu-sdk/tree/recipes-graphics/imx-gpu-sdk/imx-gpu-sdk-tesselation-rle_1.0.0.bb

That all said, there is actually an issue still with this implementation. Given the minimal usage by this upstreaming, it may be in our best interests to withdraw this from consideration for now.

Reply:
I'd have to copy these classes into multiple recipes, so I changed my mind about withdrawing this, It would be a lot of work for minimal gain, at the expense of complicating multiple recipes.